narwhals/_pandas_like/utils.py

```diff
@@ -674,12 +673,10 @@ def pivot_table(
     columns: Sequence[str],
     aggregate_function: str | None,
 ) -> Any:
-    dtypes = import_dtypes_module(df._version)
+    categorical = df._version.dtypes.Categorical
     if df._implementation is Implementation.CUDF:
-        if any(
-            x == dtypes.Categorical
-            for x in df.simple_select(*[*values, *index, *columns]).schema.values()
-        ):
+        dtypes = df.simple_select(*chain(values, index, columns)).schema.values()
+        if any(isinstance(x, categorical) for x in dtypes):
```

Comment:
I know it's fairly unrelated from the core of this PR, but we might also skip the additional select statement as all we care about are dtypes:

```suggestion
        dtypes = (
            dtype for key, dtype in df.schema.items()
            if key in set(chain(values, index, columns)))
        )
        if any(isinstance(x, categorical) for x in dtypes):
```

Reply:
Will try this out!

Reply:
Yeah nice one @FBruzzesi!

Tweaked it some more, settling on (https://github.com/narwhals-dev/narwhals/pull/2431/commits/146011d4e7008152c0a487c2596e1dc8b5b869c8)